src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -38,13 +44,18 @@ public JsonResult execute()
         AccountRequest accountRequest;
 
         try {
-            accountRequest = sqlLogic.createAccountRequest(instructorName, instructorEmail, instructorInstitution,
-                    AccountRequestStatus.PENDING, comments);
-            taskQueuer.scheduleAccountRequestForSearchIndexing(instructorEmail, instructorInstitution);
+            accountRequest = sqlLogic.createAccountRequestWithTransaction(instructorName, instructorEmail,
+                    instructorInstitution, AccountRequestStatus.PENDING, comments);
         } catch (InvalidParametersException ipe) {
             throw new InvalidHttpRequestBodyException(ipe);
         }
 
+        taskQueuer.scheduleAccountRequestForSearchIndexing(instructorEmail, instructorInstitution);
+
+        if (accountRequest.getRegisteredAt() != null) {
+            throw new InvalidOperationException("Cannot create account request as instructor has already registered.");
+        }
```

Comment:
Are we still allowing people to create account requests no matter what (except for invalid parameters)? If so, this may need to be removed.

Reply:
Yes you are right that account request is created without `registeredAt`. Have updated it. Thanks!